presto-main/src/test/java/com/facebook/presto/operator/BenchmarkHashBuildAndJoinOperators.java

```diff
@@ -207,15 +206,9 @@ public void setup()
                     throw new UnsupportedOperationException(format("Unknown outputColumns value [%s]", hashColumns));
             }
 
-            lookupSourceFactory = new BenchmarkHashBuildAndJoinOperators().benchmarkBuildHash(this, outputChannels);
             initializeProbePages();
```

Comment:
What about create `JoinOperatorFactory` in `JoinContext.setup`. So finish the build side won't destroy the join bridge prematurely. 

Some method extraction done in this PR might no longer be necessary if we go this way.

Reply:
Assuming I understood correctly, I'm having trouble with this:

```
Exception in thread "main" java.lang.IllegalStateException: Counter has been read
	at com.google.common.base.Preconditions.checkState(Preconditions.java:504)
	at com.facebook.presto.operator.JoinBridgeManager$FreezeOnReadCounter.increment(JoinBridgeManager.java:528)
	at com.facebook.presto.operator.JoinBridgeManager.incrementProbeFactoryCount(JoinBridgeManager.java:100)
```

This is because `JoinOperatorFactory` needs the `LookupSourceFactory`, which is obtained from `benchmarkBuildHash`.  I think I would need similar method extraction to make it so that `JoinOperatorFactory` is constructed before the hash operator is built (if it happens the other way around you have the above error).